packages/investor-foxy/src/api/api.ts

```diff
@@ -0,0 +1,856 @@
+      to: contractAddress,
+      value: '0'
+    }
+    const txToSign = buildTxToSign(payload)
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await this.adapter.signTransaction({ txToSign, wallet })
+      if (dryRun) return signedTx
+      return this.broadcastTx(signedTx)
+    } else if (wallet.supportsBroadcast() && this.adapter.signAndBroadcastTransaction) {
+      if (dryRun) {
+        throw new Error(`Cannot perform a dry run with wallet of type ${wallet.getVendor()}`)
+      }
+      return this.adapter.signAndBroadcastTransaction({ txToSign, wallet })
+    } else {
+      throw new Error('Invalid HDWallet configuration ')
+    }
+  }
+
+  async sendWithdrawalRequests(input: TxInputWithoutAmount): Promise<string> {
+    const { accountNumber = 0, dryRun = false, contractAddress, userAddress, wallet } = input
+    this.verifyAddresses([userAddress, contractAddress])
+    if (!wallet || !contractAddress) throw new Error('Missing inputs')
+
+    let estimatedGasBN: BigNumber
+    try {
+      estimatedGasBN = await this.estimateSendWithdrawalRequestsGas(input)
+    } catch (e) {
+      throw new Error(`Estimate Gas Error: ${e}`)
+    }
+
+    const stakingContract = this.foxyStakingContracts.find(
+      (item) => toLower(item.options.address) === toLower(contractAddress)
+    )
+    if (!stakingContract) throw new Error('Not a valid contract address')
+
+    // TODO: check if can sendWithdrawalRequests and throw an error if can't
+
+    const data: string = stakingContract.methods.sendWithdrawalRequests().encodeABI({
+      from: userAddress
+    })
+
+    let nonce: number
+    try {
+      nonce = await this.web3.eth.getTransactionCount(userAddress)
+    } catch (e) {
+      throw new Error(`Get nonce Error: ${e}`)
+    }
+    let gasPrice: string
+    try {
+      gasPrice = await this.web3.eth.getGasPrice()
+    } catch (e) {
+      throw new Error(`Get gasPrice Error: ${e}`)
+    }
+    const estimatedGas = estimatedGasBN.toString()
+    const bip44Params = this.adapter.buildBIP44Params({ accountNumber })
+    const chainId = Number(this.network)
+    const payload = {
+      bip44Params,
+      chainId,
+      data,
+      estimatedGas,
+      gasPrice,
+      nonce: String(nonce),
+      to: contractAddress,
+      value: '0'
+    }
+    const txToSign = buildTxToSign(payload)
+    if (wallet.supportsOfflineSigning()) {
+      const signedTx = await this.adapter.signTransaction({ txToSign, wallet })
+      if (dryRun) return signedTx
+      return this.broadcastTx(signedTx)
+    } else if (wallet.supportsBroadcast() && this.adapter.signAndBroadcastTransaction) {
+      if (dryRun) {
+        throw new Error(`Cannot perform a dry run with wallet of type ${wallet.getVendor()}`)
+      }
+      return this.adapter.signAndBroadcastTransaction({ txToSign, wallet })
+    } else {
+      throw new Error('Invalid HDWallet configuration ')
+    }
+  }
+
+  async addLiquidity(input: TxInput): Promise<string> {
```

Comment:
comment as this isn't really ever going to be a user facing function, and is more of a util for devs to be able to add/remove liquidity